src/Gizmos/gizmo.ts

```diff
@@ -188,7 +196,7 @@ export class Gizmo implements IDisposable {
                 effectiveNode.getWorldMatrix().decompose(undefined, this._rootMesh.rotationQuaternion!);
             }
             else {
-                this._rootMesh.rotationQuaternion!.set(0, 0, 0, 1);
+                this._rootMesh.rotationQuaternion!.copyFrom(this._customRotationQuaternion || new Quaternion(0, 0, 0, 1));

```

Comment:
I think the `_update` function may be called often (each frame?), so you should avoid creating a new object each time.

Maybe:
```typescript
if (this._customRotationQuaternion) {
    this._rootMesh.rotationQuaternion!.copyFrom(this._customRotationQuaternion);
} else {
    this._rootMesh.rotationQuaternion!.set(0, 0, 0, 1);
}
```

Reply:
how about this:

`this._rootMesh.rotationQuaternion!.set(...(this._customRotationQuaternion?.asArray() || [0, 0, 0, 1]));`

i think the if branch makes the code look redundant,

and,literal representations can be more efficient than object representations

Reply:
No, that won't be better because `asArray` is creating a new array each time you call it.

I don't see a way to make it faster than calling `copyFrom` / `set` as in my snippet because those two methods simply take the parameters and put them in the object.